lib/modules/showImages.js

```diff
@@ -472,11 +472,15 @@ addModule('showImages', (module, moduleID) => {
 	}
 
 	function updateRevealedImages(type) {
-		for (const image of imageList) {
-			if ($(image).hasClass(type) || image.imageLink.expandOnViewAll) {
-				revealImage(image, findImageFilter(image.imageLink));
-			}
+		let images = imageList.filter(v => $(v).hasClass(type) || v.imageLink.expandOnViewAll);
+
+		if (RESUtils.pageType() === 'linklist') {
+			images = images.map(v => [$(v).closest('.thing').data('rank'), v])
```

Comment:
Or even just change this line to
```js
images = images.map(v => [$(v).offset().top, v])
```

N is going to be small (<1000) in almost all cases, so I don't think there'd be a noticeable gain from sorting them up front. At least not enough to justify the complexity, IMO.